src/query.ts

```diff
@@ -111,6 +142,58 @@ async function run(): Promise<void> {
   }
 }
 
+async function uploadRepoResultToActions(
+  runQueryResult: RunQueryResult,
+  artifactClient: ArtifactClient,
+  repo: Repo
+) {
+  const filesToUpload = [
+    runQueryResult.bqrsFilePath,
+    runQueryResult.metadataFilePath,
+  ];
+  if (runQueryResult.sarifFilePath) {
+    filesToUpload.push(runQueryResult.sarifFilePath);
+  }
+  console.log("Uploading artifact");
+  await artifactClient.uploadArtifact(
+    repo.id.toString(),
+    filesToUpload,
+    "results",
+    { continueOnError: false }
+  );
+}
+
+async function uploadRepoResult(
+  controllerRepoId: number,
+  variantAnalysisId: number,
+  repo: Repo,
+  runQueryResult: RunQueryResult
+) {
+  // Get policy for artifact upload
+  const policy = await getPolicyForRepoArtifact(
+    controllerRepoId,
+    variantAnalysisId,
+    repo.id,
+    runQueryResult.sarifFileSize || runQueryResult.bqrsFileSize
+  );
+
+  // Use Azure client for uploading to Azure Blob Storage
+  const fileToUpload =
+    runQueryResult.sarifFilePath || runQueryResult.bqrsFilePath;
+  await uploadArtifact(policy, fileToUpload);
```

Comment:
I've spotted here that we were uploading the raw sarif file as the artifact, when I believe what we want to do is upload a zip file containing the sarif file or the bqrs file. I've had to add a new library to do this (https://github.com/Stuk/jszip) but hopefully this is fine. See 7318317d4bf39e649e792e380d647b8552df99b3 and 322551801e1d9ac6fb1f33711dca0f677392a7af